code-challenges.test.js

```diff
@@ -57,6 +98,34 @@ var hand2 = [5, 5, 3, 3, 4]
 var hand3 = [5, 5, 5, 5, 4]
 // Expected output: false
 
-
+describe("fullHouse", () => {
+    it("function that takes in an array of 5 numbers and determines whether or not the array is a full house", () => {
+        expect(fullHouse([5, 5, 5, 3, 3])).toEqual(true)
+        expect(fullHouse([5, 5, 3, 3, 4])).toEqual(false)
+        expect(fullHouse([5, 5, 5, 5, 4])).toEqual(false)
+    })
+})
 
 // b) Create the function that makes the test pass.
+
+
+const fullHouse = (array) => {
+    // Create a function called fullHouse that takes in an array as a parameter
+    let houseArray=[]
+    // Create an empty array to push the iterated return to
+    const counts = {};
+    // Create an object called counts to show what the count is for each index
+    for (const num of array) {
+      counts[num] = counts[num] ? counts[num] + 1 : 1
+    } // Create a loop with for...of to iterate over the array to compare counts with a ternary operator
+    for(const key in counts){
+        // Us for...in to count the number of instances for each value in the object
+        houseArray.push(counts[key])
+        // push the key and the counts for each value to the empty houseArray
+    }
+    for(let i=0;i<houseArray.length;i++){
+      if(houseArray[i]===3)return true
+      // Use for loop to iterate through the houseArray, if one value is only returned three times, return true
+    }return false
+    // else return false
+}
```

Comment:
You did a great job at explaining the process of logic here. I did find an edge case that broke the logic 
```
> console.log(fullHouse([5,5,5,2,1]))
> true
```

I think this can be fixed by adding an extra conditional after line 127.